### PR TITLE
PDF Annotations: Documentation and integration guide

### PR DESCRIPTION
**Pull request by chrisb-100ms**

### CODE REVIEW

File: docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  1. Invalid local PDF file.
+  2. Invalid or non-PDF URL.
+
+Check out the code snippet in our repository to see how we built this. [View Snippet](https://github.com/100mslive/100ms-web/blob/main/src/components/pdfAnnotator/pdfFileOptions.jsx)
+
+### Render and Share the PDF
+
+![Embed PDF Viewer](/docs/v2/pdf-sharing/pdf-sharing-embed.png)
+
+Finally, add an iframe element that points to the PDF viewer and start screen sharing. You can refer to the code snippet and comments below:
```
[the remark this answers is not in the document]
Comment (arjunattam) [reply docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx:56]: maybe we could add a section on the top that talks about how this works? a small schematic that shows iframe, screen sharing and hosted pdf viewer - and how they connect together?

File: docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx [hunk 2]
```diff
@@ -0,0 +1,196 @@
+---
+title: Share and Annotate PDFs
+nav: 1.35
+---
+
+You can now enable PDF annotations and sharing, allowing users to share and annotate PDFs in real-time, providing a rich immersive experience without tab switching.
```
Comment (arjunattam) [inline docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx:6]: would be good to call out that a shared PDF with annotations will be visible on all platforms (as a screen share)

Comment (arjunattam) [inline docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx:6]: also good to call out it's availalbe in chrome only. and maybe a reason on why that is the case?

Comment (chrisb-100ms) [reply docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx:6]: Not chrome only, works on other browsers - only region capture isn't available on other browsers so the viewer experience is = a tab being screen shared

File: docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+import { Box, ThemeTypes, useTheme } from "@100mslive/react-ui";
+
+/**
+ * PDFEmbedComponent is responsible for rendering the PDF iframe and managing the screen sharing functionality.
+ * It receives the pdfConfig and setPDFConfig as props for accessing and updating the PDF configuration state.
+ * 
+ * pdfConfig ({isPdfBeingShared: boolean})
+ */
+export const PDFEmbedComponent = ({ pdfConfig, setPDFConfig }) => {
+  let pdfIframeURL = process.env.REACT_APP_PDFJS_IFRAME_URL;
```
Comment (saikatmitra91) [inline docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx:70]: should we use process environment variable ? 

also why expose PDFJS as name ?

File: docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+  // Check if the current browser is Chrome to force current tab selection
+  const isChrome = parsedUserAgent.getBrowser()?.name?.toLowerCase() === "chrome";
```
Comment (saikatmitra91) [inline docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx:73]: this code will break as there is no reference for `parsedUserAgent`

Comment (saikatmitra91) [reply docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx:73]: we should ideally use ua-parser

File: docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+  const pdfIframeRef = useRef(); // Create a ref to access the iframe element
+  const themeType = useTheme().themeType; // Get the current theme type from the theme context
```
Comment (saikatmitra91) [inline docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx:75]: dont we have defaults for theme ? 

is this needed ?

File: docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+  const [isPDFLoaded, setIsPDFLoaded] = useState(false);
+  const { amIScreenSharing, toggleScreenShare } =
+    useScreenShare(throwErrorHandler);
```
Comment (saikatmitra91) [inline docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx:79]: this is not a right abstraction ... if someone needs to pass `throwErrorHandler` .. why cant we just abstract it ?

File: docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+  const [wasScreenShared, setWasScreenShared] = useState(false);
+
+  // If a PDF URL is provided instead of a file, update the PDF iframe URL with the URL parameter
+  if (pdfConfig.url && !pdfConfig.file) {
```
Comment (saikatmitra91) [inline docs/javascript/v2/how-to-guides/enable-content-sharing/pdf-annotations.mdx:83]: all of this should be inside a custom hook. we shouldnt expose this

**Review by saikatmitra91 — CHANGES_REQUESTED** [review]

**Pull request merged**